lib/decompress.ps1

```diff
@@ -123,15 +123,21 @@ function Expand-7zipArchive {
     }
     if (!$IsTar -and $ExtractDir) {
         movedir "$DestinationPath\$ExtractDir" $DestinationPath | Out-Null
-        # Remove temporary directory
-        Remove-Item "$DestinationPath\$($ExtractDir -replace '[\\/].*')" -Recurse -Force -ErrorAction Ignore
+        # Remove temporary directory if it still exists and is empty
+        $ExtractDirFullPath = [string] "$DestinationPath\$($ExtractDir -replace '[\\/].*')"
+        if (
+            (Test-Path -Path $ExtractDirFullPath -PathType 'Container') -and
+            (Get-ChildItem -Path $ExtractDirFullPath -Force).Count -eq 0
+        ) {
+            Remove-Item -Path $ExtractDirFullPath -Recurse -Force -ErrorAction Ignore
+        }
```

Comment:
```suggestion
        $ExtractDirTopPath = "$DestinationPath\$($ExtractDir -replace '[\\/].*')"
        if ((Get-ChildItem -Path $ExtractDirTopPath -Force -ErrorAction Ignore).Count -eq 0) {
            Remove-Item -Path $ExtractDirTopPath -Recurse -Force -ErrorAction Ignore
        }
```

Change param name, and using `-ErrorAction Ignore` in `Get-ChildItem` will solve the prob (return $null that `Count` is 0 if Path is not existed).

Reply:
That's just up to preference, but I'll comply with whatever Scoop wants.